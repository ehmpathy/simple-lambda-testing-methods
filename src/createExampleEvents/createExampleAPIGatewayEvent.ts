import { APIGatewayEvent } from 'aws-lambda';

import { EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED } from './eventMetadataWhichShouldNotBeNeeded';

export const createExampleAPIGatewayEvent = ({
  body,
  httpMethod,
  isBase64Encoded = false,
  headers = {
    'Content-Type': 'application/json',
  },
  path = '__path__',
  pathParameters = null,
  queryStringParameters = null,
}: {
  body: string;
  httpMethod: 'POST' | 'GET';
  /**
   * aws will tell us whether they think the body is a base64 encoded binary or not, based on the headers
   *
   * ref: https://aws.amazon.com/blogs/compute/handling-binary-data-using-amazon-api-gateway-http-apis/
   */
  isBase64Encoded?: boolean;
  headers?: Record<string, string>;
  path?: string;
  pathParameters?: Record<string, string> | null;
  queryStringParameters?: Record<string, string> | null;
}): APIGatewayEvent => {
  return {
    path,
    pathParameters,
    queryStringParameters,
    multiValueQueryStringParameters: queryStringParameters
      ? Object.fromEntries(
          Object.entries(queryStringParameters).map(([key, value]) => [
            key,
            [value],
          ]),
        )
      : null,
    body,
    headers,
    multiValueHeaders: Object.fromEntries(
      Object.entries(headers).map(([key, value]) => [key, [value]]),
    ),
    httpMethod,
    isBase64Encoded,
    stageVariables: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
    requestContext: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
    resource: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
  };
};
