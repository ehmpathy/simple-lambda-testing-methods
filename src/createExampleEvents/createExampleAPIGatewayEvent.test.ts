import { APIGatewayEvent } from 'aws-lambda';

import { createExampleAPIGatewayEvent } from './createExampleAPIGatewayEvent';

describe('createExampleAPIGatewayEvent', () => {
  it('should create an accurate looking sns event', () => {
    const event: APIGatewayEvent = createExampleAPIGatewayEvent({
      body: JSON.stringify({ hello: 'there' }),
      httpMethod: 'POST',
    });
    expect(event).toMatchSnapshot();
  });
});
