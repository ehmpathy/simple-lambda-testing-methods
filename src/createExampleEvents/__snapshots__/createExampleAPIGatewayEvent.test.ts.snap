// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExampleAPIGatewayEvent should create an accurate looking sns event 1`] = `
Object {
  "body": "{\\"hello\\":\\"there\\"}",
  "headers": Object {
    "Content-Type": "application/json",
  },
  "httpMethod": "POST",
  "isBase64Encoded": false,
  "multiValueHeaders": Object {
    "Content-Type": Array [
      "application/json",
    ],
  },
  "multiValueQueryStringParameters": null,
  "path": "__path__",
  "pathParameters": null,
  "queryStringParameters": null,
  "requestContext": undefined,
  "resource": undefined,
  "stageVariables": undefined,
}
`;
